packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMainInput.tsx

```diff
@@ -140,34 +148,89 @@ export type TransferPanelMainInputProps =
     maxButtonOnClick: React.ButtonHTMLAttributes<HTMLButtonElement>['onClick']
     value: string
     tokenButtonOptions?: TokenButtonOptions
+    maxAmount: string | undefined
+    isMaxAmount: boolean
   }
 
-export function TransferPanelMainInput(props: TransferPanelMainInputProps) {
-  const { errorMessage, maxButtonOnClick, tokenButtonOptions, ...rest } = props
+export const TransferPanelMainInput = React.memo(
+  ({
+    errorMessage,
+    maxButtonOnClick,
+    tokenButtonOptions,
+    onChange,
+    maxAmount,
+    value,
+    isMaxAmount,
+    ...rest
+  }: TransferPanelMainInputProps) => {
+    const [localValue, setLocalValue] = useState(value)
+    const decimals = useSelectedTokenDecimals()
```

Comment:
this should be based on the token that this `TransferPanelMainInput` is for, otherwise it would break for batch transfer because the two tokens can have different number of decimals